### PR TITLE
[DDW-677] Implement alternative update path for manual updates

### PR DESCRIPTION
**Pull request by tomothespian**
This PR introduces alternative update path for manual updates when app is out of date and stuck on connecting or syncing status longer than 5 minutes.

## Screenshots

_Storybook story_

Storybook - Manual update overlay

_In-App implementation_
EN - Manual App Update overlay

---

## Testing Checklist

- [Slack QA thread](https://input-output-rnd.slack.com/archives/GGKFXSKC6/p1559204209009200)
- [ ] Test Storybook story under: `ManualUpdateOverlay > Defaults`

---

### Basics

- [ ] PR has been assigned and has appropriate labels (`feature`/`bug`/`chore`, `release-x.x.x`)
- [ ] PR is updated to the most recent version of the target branch (and there are no conflicts)
- [ ] PR has a good description that summarizes all changes and shows some screenshots or animated GIFs of important UI changes
- [ ] CHANGELOG entry has been added to the top of the appropriate section (*Features*, *Fixes*, *Chores*) and is linked to the correct PR on GitHub
- [ ] Automated tests: All acceptance and unit tests are passing (`yarn test`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *development* build (`yarn dev`)
- [ ] Manual tests (minimum tests should cover newly added feature/fix): App works correctly in *production* build (`yarn package` / CI builds)
- [ ] There are no *flow* errors or warnings (`yarn flow:test`)
- [ ] There are no *lint* errors or warnings (`yarn lint`)
- [ ] There are no *prettier* errors or warnings (`yarn prettier:check`)
- [ ] There are no missing translations (running `yarn manage:translations` produces no changes)
- [ ] Text changes are proofread and approved (Jane Wild / Amy Reeve)
- [ ] Japanese text changes are proofread and approved (Junko Oda)
- [ ] UI changes look good in all themes (Alexander Rukin)
- [ ] Storybook works and no stories are broken (`yarn storybook`)
- [ ] In case of dependency changes `yarn.lock` file is updated

### Code Quality
- [ ] Important parts of the code are properly commented and documented
- [ ] Code is properly typed with flow
- [ ] React components are split-up enough to avoid unnecessary re-renderings
- [ ] Any code that only works in main process is neatly separated from components

### Testing
- [ ] New feature/change is covered by acceptance tests
- [ ] New feature/change is manually tested and approved by QA team
- [ ] All existing acceptance tests are still up-to-date
- [ ] New feature/change is covered by Daedalus Testing scenario
- [ ] All existing Daedalus Testing scenarios are still up-to-date

### After Review
- [ ] Merge the PR
- [ ] Delete the source branch
- [ ] Move the ticket to `done` column on the YouTrack board
- [ ] Update Slack QA thread by marking it with a green checkmark

### CODE REVIEW

File: source/renderer/app/components/loading/ManualUpdateOverlay.scss [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  position: fixed;
+  right: 0;
+  top: 0;
+  z-index: 9999;
+
+  .dialog {
+    margin: auto 0;
+    overflow: hidden;
+    width: 100%;
+    height: 100%;
```
Comment (nikolaglumac) [inline source/renderer/app/components/loading/ManualUpdateOverlay.scss]: @tomothespian please sort CSS rules alphabetically 🙏

File: source/renderer/app/components/loading/ManualUpdateOverlay.scss [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+    &:focus {
+      outline: none;
+    }
+    .content {
+      background-color: var(--theme-manual-update-overlay-background-color);
+      display: flex;
+      font-family: var(--font-regular);
+      flex-direction: column;
+      height: 100%;
```
Comment (nikolaglumac) [inline source/renderer/app/components/loading/ManualUpdateOverlay.scss]: @tomothespian please sort CSS rules alphabetically 🙏

File: source/renderer/app/components/loading/ManualUpdateOverlay.scss [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        margin: 16px auto;
+        p {
+          color: var(--theme-manual-update-overlay-text-color);
+          font-size: 16px;
+          line-height: 1.38;
+          margin-bottom: 14px;
+          max-width: 610px;
+          b {
+            font-family: var(--font-medium);
+            color: var(--theme-manual-update-overlay-text-highlight-color);
```
Comment (nikolaglumac) [inline source/renderer/app/components/loading/ManualUpdateOverlay.scss]: @tomothespian please sort CSS rules alphabetically 🙏

File: source/renderer/app/components/loading/ManualUpdateOverlay.scss [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+          }
+        }
+      }
+
+      .actionButton {
+        background-color: var(--theme-manual-update-overlay-button-background-color);
+        border: solid 1px var(--theme-manual-update-overlay-button-border-color);
+        margin: 0 auto;
```
Comment (nikolaglumac) [inline source/renderer/app/components/loading/ManualUpdateOverlay.scss]: @tomothespian please sort CSS rules alphabetically 🙏

File: source/renderer/app/components/loading/ManualUpdateOverlay.scss [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+          width: 12px;
+          margin-left: 8px;
+          vertical-align: middle;
+          opacity: 0.7;
+        }
+     
+        &:not(.disabled):hover {
+          border: none;
+          color: var(--theme-manual-update-overlay-button-text-color-hover);
+          background-color: var(
```
Comment (nikolaglumac) [inline source/renderer/app/components/loading/ManualUpdateOverlay.scss]: @tomothespian please sort CSS rules alphabetically 🙏

File: source/renderer/app/containers/LoadingPage.js [hunk 6]
```diff
@@ -146,4 +158,13 @@ export default class LoadingPage extends Component<InjectedProps> {
       profile.downloadLogs.trigger({ fileName, destination, fresh: true });
     }
   };
+
+  handleGetAvailableVersions = () => {
+    const { app } = this.props.actions;
+    app.getLatestAvailableAppVersion.trigger();
+  };
+
+  handleManualUpdateInstructionsLinkClick = (url: string) => {
```
[the remark this answers is not in the document]
Comment (tomothespian) [reply source/renderer/app/containers/LoadingPage.js]: @daniloprates nice hint, thanks.

File: source/renderer/app/stores/AppStore.js [hunk 7]
```diff
@@ -25,6 +26,9 @@ export default class AppStore extends Store {
   @observable gpuStatus: ?GpuStatus = null;
   @observable numberOfEpochsConsolidated: number = 0;
   @observable previousRoute: string = ROUTES.ROOT;
+  @observable availableAppVersion: ?string = null;
```
Comment (nikolaglumac) [inline source/renderer/app/stores/AppStore.js]: @tomothespian I feel like these app version related observables and actions should be moved into the `NodeUpdateStore`. Can you please move them?

File: source/renderer/app/stores/NodeUpdateStore.js [hunk 8]
```diff
@@ -57,4 +62,21 @@ export default class NodeUpdateStore extends Store {
...
+  @action _getLatestAvailableAppVersion = async () => {
+    const { version, platform, isDevelopment } = this.environment;
+
+    if (!isDevelopment) {
+      this.isNewAppVersionLoading = true;
+      const versionInfo = await this.api.ada.getLatestAppVersionInfo();
+      const availableVersion = get(versionInfo, ['platforms', platform, 'version'], null);
+      const isNewAppVersionAvailable = availableVersion && availableVersion > version;
+
+      runInAction(() => {
```
Comment (nikolaglumac) [inline source/renderer/app/stores/NodeUpdateStore.js]: @tomothespian please add a name for this "run in action"

File: source/renderer/app/stores/NodeUpdateStore.js [hunk 9]
```diff
@@ -57,4 +62,21 @@ export default class NodeUpdateStore extends Store {
...
   };
 
+  @action _getLatestAvailableAppVersion = async () => {
+    const { version, platform, isDevelopment } = this.environment;
+
+    if (!isDevelopment) {
+      this.isNewAppVersionLoading = true;
+      const versionInfo = await this.api.ada.getLatestAppVersionInfo();
+      const availableVersion = get(versionInfo, ['platforms', platform, 'version'], null);
+      const isNewAppVersionAvailable = availableVersion && availableVersion > version;
```
Comment (nikolaglumac) [inline source/renderer/app/stores/NodeUpdateStore.js]: @tomothespian are you sure this comparison works? Shouldn't we compare all 3 different segments of the version? E.g. individual major, middle and minor version checks?

**Comment by nikolaglumac** [comment]
This feature is to be released in Daedalus `0.14.0` as the `0.13.2` release has been canceled...

**Comment by nikolaglumac** [comment]
CHANGELOG cleanup task: https://iohk.myjetbrains.com/youtrack/issue/DDW-720

**Comment by nikolaglumac** [comment]
Version 0.13.2 removed from the `CHANGELOG` and `package.jsonà in: https://github.com/input-output-hk/daedalus/pull/1441

**Pull request merged**